core/branch/branch.go

```diff
@@ -0,0 +1,18 @@
+// Package branch contains the core branch service interface.
+package branch
+
+import "context"
+
+// Service is the branch service interface. It can be used to execute 
+// code paths in an isolated execution context that can be reverted.
```

Comment:
Let's say more about what this means. What is isolated and what gets reverted?

Reply:
addressed